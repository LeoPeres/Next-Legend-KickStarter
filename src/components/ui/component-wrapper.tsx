import React from "react";
import { cn } from "@/lib/utils";

interface ComponentWrapperProps extends React.HTMLAttributes<HTMLDivElement> {
  title?: string;
  description?: string;
  children: React.ReactNode;
  className?: string;
  contentClassName?: string;
  headerClassName?: string;
  showHeader?: boolean;
}

/**
 * ComponentWrapper provides a consistent container for UI components
 * with optional title and description.
 *
 * @example
 * <ComponentWrapper
 *   title="User Profile"
 *   description="Manage your account settings and preferences."
 * >
 *   <ProfileForm />
 * </ComponentWrapper>
 */
export function ComponentWrapper({
  title,
  description,
  children,
  className,
  contentClassName,
  headerClassName,
  showHeader = true,
  ...props
}: ComponentWrapperProps) {
  return (
    <div className={cn("w-full", className)} {...props}>
      {showHeader && (title || description) && (
        <div className={cn("mb-4", headerClassName)}>
          {title && <h3 className="text-lg font-medium">{title}</h3>}
          {description && <p className="mt-1 text-sm text-muted-foreground">{description}</p>}
        </div>
      )}
      <div className={cn("w-full", contentClassName)}>{children}</div>
    </div>
  );
}

/**
 * FormComponentWrapper provides a consistent container specifically for form components
 * with label, description, and error handling.
 *
 * @example
 * <FormComponentWrapper
 *   label="Email Address"
 *   description="We'll never share your email with anyone else."
 *   error="Please enter a valid email address"
 * >
 *   <Input placeholder="Enter your email" />
 * </FormComponentWrapper>
 */
export function FormComponentWrapper({
  label,
  description,
  error,
  children,
  className,
  required = false,
  ...props
}: {
  label?: string;
  description?: string;
  error?: string;
  children: React.ReactNode;
  className?: string;
  required?: boolean;
}) {
  const id = React.useId();

  return (
    <div className={cn("w-full space-y-2", className)} {...props}>
      {label && (
        <label
          htmlFor={id}
          className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        >
          {label}
          {required && <span className="text-destructive ml-1">*</span>}
        </label>
      )}
      {description && <p className="text-xs text-muted-foreground">{description}</p>}
      {React.isValidElement(children)
        ? React.cloneElement(children as React.ReactElement<any>, {
            id,
            "aria-describedby": description ? `${id}-description` : undefined,
            "aria-invalid": error ? true : undefined,
          })
        : children}
      {error && (
        <p className="text-xs text-destructive" id={`${id}-error`}>
          {error}
        </p>
      )}
    </div>
  );
}
