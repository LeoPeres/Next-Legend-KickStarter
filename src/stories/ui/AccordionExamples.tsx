import React from "react";
import {
  Accordion,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
} from "../../../components/ui/accordion";

export const SingleAccordion = () => (
  <Accordion type="single" collapsible className="w-full max-w-md">
    <AccordionItem value="item-1">
      <AccordionTrigger>Is it accessible?</AccordionTrigger>
      <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2">
      <AccordionTrigger>Is it styled?</AccordionTrigger>
      <AccordionContent>
        Yes. It comes with default styles that match the other components' aesthetic.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-3">
      <AccordionTrigger>Is it animated?</AccordionTrigger>
      <AccordionContent>
        Yes. It's animated by default, but you can disable it if you prefer.
      </AccordionContent>
    </AccordionItem>
  </Accordion>
);

export const MultipleAccordion = () => (
  <Accordion type="multiple" className="w-full max-w-md">
    <AccordionItem value="item-1">
      <AccordionTrigger>First section</AccordionTrigger>
      <AccordionContent>This is the first section content.</AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2">
      <AccordionTrigger>Second section</AccordionTrigger>
      <AccordionContent>This is the second section content.</AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-3">
      <AccordionTrigger>Third section</AccordionTrigger>
      <AccordionContent>This is the third section content.</AccordionContent>
    </AccordionItem>
  </Accordion>
);

export const CustomStyledAccordion = () => (
  <Accordion type="single" collapsible className="w-full max-w-md border rounded-lg">
    <AccordionItem value="item-1" className="border-b-0 px-4">
      <AccordionTrigger className="py-6 text-blue-600">Custom styled trigger</AccordionTrigger>
      <AccordionContent className="text-gray-600">
        This accordion has custom styling applied to it.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2" className="border-b-0 px-4">
      <AccordionTrigger className="py-6 text-blue-600">Another custom section</AccordionTrigger>
      <AccordionContent className="text-gray-600">
        You can customize the appearance of the accordion to match your design system.
      </AccordionContent>
    </AccordionItem>
  </Accordion>
);

export const AccordionExamples = () => (
  <div className="space-y-8">
    <div>
      <h3 className="text-lg font-medium mb-2">Single Accordion (Collapsible)</h3>
      <SingleAccordion />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Multiple Accordion</h3>
      <MultipleAccordion />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Custom Styled Accordion</h3>
      <CustomStyledAccordion />
    </div>
  </div>
);
