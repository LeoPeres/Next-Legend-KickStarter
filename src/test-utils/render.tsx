import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// Import providers
import { ThemeProvider } from "../components/theme-provider";

// Mock the ThemeProvider to avoid issues with matchMedia and localStorage
jest.mock("../components/theme-provider", () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => <>{children}</>,
}));

// Add any providers that components need here
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) => {
  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: AllTheProviders, ...options }),
  };
};

// Re-export everything from testing-library
export * from "@testing-library/react";

// Override render method
export { customRender as render };
