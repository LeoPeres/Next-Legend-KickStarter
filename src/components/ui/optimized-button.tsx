"use client";

import React, { useState, useRef, useCallback } from "react";
import { Button, ButtonProps } from "@/components/ui/button";
import { useOptimizedClick } from "@/hooks/use-optimized-event";
import { cn } from "@/lib/utils";
import { measureInteraction } from "@/lib/performance";

interface OptimizedButtonProps extends ButtonProps {
  /**
   * Optional feedback color when button is clicked
   */
  feedbackColor?: string;

  /**
   * Whether to show a ripple effect on click
   */
  showRipple?: boolean;

  /**
   * Name for performance measurement
   */
  measureName?: string;
}

/**
 * OptimizedButton component
 *
 * An enhanced button component that improves interaction responsiveness
 * for better FID/INP metrics by:
 *
 * 1. Using optimized event listeners
 * 2. Providing visual feedback immediately on interaction
 * 3. Using CSS properties that don't trigger layout
 * 4. Measuring interaction performance
 */
export function OptimizedButton({
  children,
  className,
  feedbackColor,
  showRipple = true,
  measureName,
  onClick,
  ...props
}: OptimizedButtonProps) {
  // State for ripple effect
  const [ripples, setRipples] = useState<Array<{ x: number; y: number; id: number }>>([]);
  const nextRippleId = useRef(0);
  const buttonRef = useRef<HTMLButtonElement>(null);

  // Handle click with optimized event listener
  const handleClick = useCallback(
    (e: Event) => {
      if (!(e instanceof MouseEvent)) return;

      // Start performance measurement if name is provided
      const endMeasurement = measureName ? measureInteraction(measureName) : null;

      // Show ripple effect if enabled
      if (showRipple && buttonRef.current) {
        const button = buttonRef.current;
        const rect = button.getBoundingClientRect();
        const x = e.clientX - rect.left;
        const y = e.clientY - rect.top;
        const id = nextRippleId.current++;

        setRipples((prev) => [...prev, { x, y, id }]);

        // Remove ripple after animation completes
        setTimeout(() => {
          setRipples((prev) => prev.filter((ripple) => ripple.id !== id));
        }, 600);
      }

      // Call the original onClick handler if provided
      if (onClick) {
        // Create a synthetic React mouse event from the native event
        // This is a simplified version that works for our purposes
        const syntheticEvent = {
          ...e,
          preventDefault: () => e.preventDefault(),
          stopPropagation: () => e.stopPropagation(),
          target: e.target,
          currentTarget: buttonRef.current,
          nativeEvent: e,
          isDefaultPrevented: () => e.defaultPrevented,
          isPropagationStopped: () => false,
          persist: () => {},
        } as unknown as React.MouseEvent<HTMLButtonElement>;

        onClick(syntheticEvent);
      }

      // End performance measurement
      if (endMeasurement) {
        endMeasurement();
      }
    },
    [onClick, showRipple, measureName]
  );

  // Use optimized click event
  useOptimizedClick(handleClick as EventListener, buttonRef.current);

  return (
    <Button
      ref={buttonRef}
      className={cn(
        "relative overflow-hidden transition-transform active:scale-[0.98]",
        "touch-none", // Prevent double-tap zoom on mobile
        className
      )}
      onClick={undefined} // We handle clicks with our optimized handler
      style={
        {
          // Use CSS variables for dynamic values to avoid layout thrashing
          "--feedback-color": feedbackColor || "rgba(255, 255, 255, 0.1)",
        } as React.CSSProperties
      }
      {...props}
    >
      {children}

      {/* Ripple effects */}
      {ripples.map((ripple) => (
        <span
          key={ripple.id}
          className="absolute rounded-full bg-white/20 pointer-events-none animate-ripple"
          style={{
            left: ripple.x,
            top: ripple.y,
            transform: "translate(-50%, -50%)",
            width: "200%",
            aspectRatio: "1/1",
          }}
        />
      ))}
    </Button>
  );
}
