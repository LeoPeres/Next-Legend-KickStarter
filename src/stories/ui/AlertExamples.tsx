import React from "react";
import { Alert, AlertTitle, AlertDescription } from "../../../components/ui/alert";
import { Terminal, AlertCircle, Info, CheckCircle, Bell } from "lucide-react";

export const DefaultAlert = () => (
  <Alert className="w-full max-w-md">
    <Terminal className="h-4 w-4" />
    <AlertTitle>Heads up!</AlertTitle>
    <AlertDescription>You can add components to your app using the CLI.</AlertDescription>
  </Alert>
);

export const DestructiveAlert = () => (
  <Alert variant="destructive" className="w-full max-w-md">
    <AlertCircle className="h-4 w-4" />
    <AlertTitle>Error</AlertTitle>
    <AlertDescription>Your session has expired. Please log in again.</AlertDescription>
  </Alert>
);

export const InfoAlert = () => (
  <Alert className="w-full max-w-md">
    <Info className="h-4 w-4" />
    <AlertTitle>Information</AlertTitle>
    <AlertDescription>This feature is currently in beta and may change.</AlertDescription>
  </Alert>
);

export const SuccessAlert = () => (
  <Alert className="w-full max-w-md border-green-500">
    <CheckCircle className="h-4 w-4 text-green-500" />
    <AlertTitle className="text-green-500">Success</AlertTitle>
    <AlertDescription>Your changes have been saved successfully.</AlertDescription>
  </Alert>
);

export const NotificationAlert = () => (
  <Alert className="w-full max-w-md">
    <Bell className="h-4 w-4" />
    <AlertTitle>Notification</AlertTitle>
    <AlertDescription>You have 3 unread messages in your inbox.</AlertDescription>
  </Alert>
);

export const AlertExamples = () => (
  <div className="space-y-6">
    <div>
      <h3 className="text-lg font-medium mb-2">Default Alert</h3>
      <DefaultAlert />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Destructive Alert</h3>
      <DestructiveAlert />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Info Alert</h3>
      <InfoAlert />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Success Alert (Custom Styled)</h3>
      <SuccessAlert />
    </div>
    <div>
      <h3 className="text-lg font-medium mb-2">Notification Alert</h3>
      <NotificationAlert />
    </div>
  </div>
);
