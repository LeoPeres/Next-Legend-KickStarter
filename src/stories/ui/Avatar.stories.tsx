import type { Meta, StoryObj } from "@storybook/react";
import { Avatar, AvatarImage, AvatarFallback } from "../../../components/ui/avatar";
import { AvatarExamples } from "./AvatarExamples";

const meta: Meta<typeof Avatar> = {
  title: "UI/Avatar",
  component: Avatar,
  parameters: {
    layout: "centered",
    docs: {
      description: {
        component: "An image element with a fallback for representing the user.",
      },
    },
  },
  argTypes: {
    className: {
      control: "text",
      description: "Additional CSS classes to apply to the avatar",
      table: {
        category: "Appearance",
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  render: () => (
    <Avatar>
      <AvatarImage src="https://github.com/shadcn.png" alt="@shadcn" />
      <AvatarFallback>CN</AvatarFallback>
    </Avatar>
  ),
};

export const WithFallback: Story = {
  render: () => (
    <Avatar>
      <AvatarImage src="/broken-image.jpg" alt="@user" />
      <AvatarFallback>JD</AvatarFallback>
    </Avatar>
  ),
};

export const CustomSize: Story = {
  render: () => (
    <Avatar className="h-16 w-16">
      <AvatarImage src="https://github.com/shadcn.png" alt="@shadcn" />
      <AvatarFallback>CN</AvatarFallback>
    </Avatar>
  ),
};

export const CustomFallbackColor: Story = {
  render: () => (
    <Avatar>
      <AvatarImage src="/broken-image.jpg" alt="@user" />
      <AvatarFallback className="bg-blue-500 text-white">JD</AvatarFallback>
    </Avatar>
  ),
};

export const CompleteExamples: Story = {
  render: () => <AvatarExamples />,
  parameters: {
    layout: "padded",
    docs: {
      description: {
        story: "Complete set of avatar examples showing various configurations.",
      },
    },
  },
};
