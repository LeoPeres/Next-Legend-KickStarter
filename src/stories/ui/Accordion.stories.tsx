import type { Meta, StoryObj } from "@storybook/react";
import {
  Accordion,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
} from "../../../components/ui/accordion";
import { AccordionExamples } from "./AccordionExamples";

const meta: Meta<typeof Accordion> = {
  title: "UI/Accordion",
  component: Accordion,
  parameters: {
    layout: "centered",
    docs: {
      description: {
        component:
          "A vertically stacked set of interactive headings that each reveal a section of content.",
      },
    },
  },
  argTypes: {
    type: {
      control: "select",
      options: ["single", "multiple"],
      description: "Determines whether one or multiple items can be opened at the same time",
      table: {
        category: "Behavior",
        defaultValue: { summary: "single" },
      },
    },
    collapsible: {
      control: "boolean",
      description:
        "When type is 'single', allows closing content when clicking the trigger for an open item",
      table: {
        category: "Behavior",
        defaultValue: { summary: "false" },
      },
    },
    defaultValue: {
      control: "text",
      description: "The value of the item to expand by default (controlled)",
      table: {
        category: "State",
      },
    },
    className: {
      control: "text",
      description: "Additional CSS classes to apply",
      table: {
        category: "Appearance",
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Accordion>;

export const Default: Story = {
  render: () => (
    <Accordion type="single" collapsible className="w-full max-w-md">
      <AccordionItem value="item-1">
        <AccordionTrigger>Is it accessible?</AccordionTrigger>
        <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-2">
        <AccordionTrigger>Is it styled?</AccordionTrigger>
        <AccordionContent>
          Yes. It comes with default styles that match the other components' aesthetic.
        </AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-3">
        <AccordionTrigger>Is it animated?</AccordionTrigger>
        <AccordionContent>
          Yes. It's animated by default, but you can disable it if you prefer.
        </AccordionContent>
      </AccordionItem>
    </Accordion>
  ),
};

export const Multiple: Story = {
  render: () => (
    <Accordion type="multiple" className="w-full max-w-md">
      <AccordionItem value="item-1">
        <AccordionTrigger>First section</AccordionTrigger>
        <AccordionContent>This is the first section content.</AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-2">
        <AccordionTrigger>Second section</AccordionTrigger>
        <AccordionContent>This is the second section content.</AccordionContent>
      </AccordionItem>
      <AccordionItem value="item-3">
        <AccordionTrigger>Third section</AccordionTrigger>
        <AccordionContent>This is the third section content.</AccordionContent>
      </AccordionItem>
    </Accordion>
  ),
};

export const CompleteExamples: Story = {
  render: () => <AccordionExamples />,
  parameters: {
    layout: "padded",
    docs: {
      description: {
        story: "Complete set of accordion examples showing various configurations.",
      },
    },
  },
};
