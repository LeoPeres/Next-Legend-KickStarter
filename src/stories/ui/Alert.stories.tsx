import type { Meta, StoryObj } from "@storybook/react";
import { Alert, AlertTitle, AlertDescription } from "../../../components/ui/alert";
import { AlertExamples } from "./AlertExamples";
import { Terminal, AlertCircle, Info } from "lucide-react";

const meta: Meta<typeof Alert> = {
  title: "UI/Alert",
  component: Alert,
  parameters: {
    layout: "centered",
    docs: {
      description: {
        component:
          "Displays a callout for user attention with different variants for different scenarios.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["default", "destructive"],
      description: "The visual style of the alert",
      table: {
        category: "Appearance",
        defaultValue: { summary: "default" },
      },
    },
    className: {
      control: "text",
      description: "Additional CSS classes to apply",
      table: {
        category: "Appearance",
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Alert>;

export const Default: Story = {
  render: () => (
    <Alert className="w-full max-w-md">
      <Terminal className="h-4 w-4" />
      <AlertTitle>Heads up!</AlertTitle>
      <AlertDescription>You can add components to your app using the CLI.</AlertDescription>
    </Alert>
  ),
};

export const Destructive: Story = {
  render: () => (
    <Alert variant="destructive" className="w-full max-w-md">
      <AlertCircle className="h-4 w-4" />
      <AlertTitle>Error</AlertTitle>
      <AlertDescription>Your session has expired. Please log in again.</AlertDescription>
    </Alert>
  ),
};

export const WithoutIcon: Story = {
  render: () => (
    <Alert className="w-full max-w-md">
      <AlertTitle>Note</AlertTitle>
      <AlertDescription>This is an alert without an icon.</AlertDescription>
    </Alert>
  ),
};

export const WithoutTitle: Story = {
  render: () => (
    <Alert className="w-full max-w-md">
      <Info className="h-4 w-4" />
      <AlertDescription>This is an alert with an icon but no title.</AlertDescription>
    </Alert>
  ),
};

export const CompleteExamples: Story = {
  render: () => <AlertExamples />,
  parameters: {
    layout: "padded",
    docs: {
      description: {
        story: "Complete set of alert examples showing various configurations.",
      },
    },
  },
};
