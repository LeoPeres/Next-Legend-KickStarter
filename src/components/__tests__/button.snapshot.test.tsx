import React from "react";
import { render } from "../../test-utils/render";
import { Button } from "../ui/button";

describe("Button Snapshots", () => {
  it("renders default button correctly", () => {
    const { container } = render(<Button>Default Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders destructive button correctly", () => {
    const { container } = render(<Button variant="destructive">Destructive Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders outline button correctly", () => {
    const { container } = render(<Button variant="outline">Outline Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders secondary button correctly", () => {
    const { container } = render(<Button variant="secondary">Secondary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders ghost button correctly", () => {
    const { container } = render(<Button variant="ghost">Ghost Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders link button correctly", () => {
    const { container } = render(<Button variant="link">Link Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders small button correctly", () => {
    const { container } = render(<Button size="sm">Small Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders large button correctly", () => {
    const { container } = render(<Button size="lg">Large Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders disabled button correctly", () => {
    const { container } = render(<Button disabled>Disabled Button</Button>);
    expect(container).toMatchSnapshot();
  });
});
