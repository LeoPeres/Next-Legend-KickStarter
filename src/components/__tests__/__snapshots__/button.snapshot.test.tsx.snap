// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots renders default button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
  >
    Default Button
  </button>
</div>
`;

exports[`Button Snapshots renders destructive button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-destructive text-destructive-foreground hover:bg-destructive/90 h-10 py-2 px-4"
  >
    Destructive Button
  </button>
</div>
`;

exports[`Button Snapshots renders disabled button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
    disabled=""
  >
    Disabled Button
  </button>
</div>
`;

exports[`Button Snapshots renders ghost button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4"
  >
    Ghost Button
  </button>
</div>
`;

exports[`Button Snapshots renders large button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground hover:bg-primary/90 h-11 px-8 rounded-md"
  >
    Large Button
  </button>
</div>
`;

exports[`Button Snapshots renders link button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none text-primary underline-offset-4 hover:underline h-10 py-2 px-4"
  >
    Link Button
  </button>
</div>
`;

exports[`Button Snapshots renders outline button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4"
  >
    Outline Button
  </button>
</div>
`;

exports[`Button Snapshots renders secondary button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 py-2 px-4"
  >
    Secondary Button
  </button>
</div>
`;

exports[`Button Snapshots renders small button correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground hover:bg-primary/90 h-9 px-3 rounded-md"
  >
    Small Button
  </button>
</div>
`;
